### ISSUE TITLE
smtplib.SMTPSenderRefused

### ISSUE THREAD

**Issue by rosdat** [CoreyMSchafer/code_snippets#52]
smtplib.SMTPSenderRefused

smtplib.SMTPSenderRefused: (530, b'5.5.1 Authentication Required. Learn more at\n5.5.1  https://support.google.com/mail/?p=WantAuthError v2sm22720640pgr.2 - gsmtp', 'example@gmail.com')

i just completed mr coreyMSchafer vedio tutorial Flask dev environment. but i get this error message.

on __init__.py already write the code :
# app.config['MAIL_SERVER'] = 'smtp.googlemail.com'
# app.config['MAIL_PORT'] = 587
# app.config['MAIL_USE_TLS'] = True
# app.config['MAIL_USERNAME'] = os.environ.get('EMAIL_USER')
# app.config['MAIL_PASSWORD'] = os.environ.get('EMAIL_PASS')

**Comment by LukasJerabek** [comment]
Same problem here...

**Comment by LukasJerabek** [comment]
Make sure "Access for less secure apps has been turned on" for your google account. Else that can cause SMTP Authentication problems. You can turn on/off from google settings: https://www.google.com/settings/security/lesssecureapps

**Comment by SangamSwadiK** [comment]
Yes, you have all permission, Access for less secure apps has been turned on.

**Comment by BadgerFool** [comment]
Did you manage to fix this issue? 
I'm getting the same.

I know the creds are fine and ""Access for less secure apps" is turned on as the code runs fine when the creds are hardcoded into config.py

The environment variables for db are working fine as I can login, create new posts etc.
The only issue is with the mail creds, which were saved in the exact same way.

I have tried on both Linux and Windows 10 as I guessed I was doing something wrong on Linux (I'm a Linux newb), however the issue persists on both operating systems.

The only other thing I have tried is changing the smtp server to 'smtp.gmail.com' as suggested [here](https://stackoverflow.com/questions/28209414/sending-a-mail-from-flask-mail-smtpsenderrefused-530) which obviously didn't work.

I have been trying to resolve this for about an hour now and obviously cannot proceed to work through the deployment tutorial with hardcoded creds :(

Any help appreciated.

**Comment by ImOmkar** [comment]
Use Sendgrid.

**Comment by maruf1847** [comment]
After Turn on less secure app.............but still now it's not working.........what i do??

**Comment by BrianMosbeux** [comment]
If you use 2-step verification with your google account you can use an App Password to solve the issue.

https://support.google.com/accounts/answer/185833?hl=en

Go to:
Google Account > Security > Signing in to Google > App passwords

There, generate a 16 characters code that you can use in your config instead of your email password

**Comment by AnandPokar** [comment]
> EDIT: RESOLVED*
> 
> Did you manage to fix this issue? I'm getting the same.
> 
> I know the creds are fine and ""Access for less secure apps" is turned on as the code runs fine when the creds are hardcoded into config.py
> 
> The environment variables for db are working fine as I can login, create new posts etc. The only issue is with the mail creds, which were saved in the exact same way.
> 
> I have tried on both Linux and Windows 10 as I guessed I was doing something wrong on Linux (I'm a Linux newb), however the issue persists on both operating systems.
> 
> The only other thing I have tried is changing the smtp server to 'smtp.gmail.com' as suggested [here](https://stackoverflow.com/questions/28209414/sending-a-mail-from-flask-mail-smtpsenderrefused-530) which obviously didn't work.
> 
> I have been trying to resolve this for about an hour now and obviously cannot proceed to work through the deployment tutorial with hardcoded creds :(
> 
> Any help appreciated.
> 
> *The problem on Windows was resolved by me using the correct env variable names 🤦‍♂️ My Linux issues are probably still related to me being a newb!

Please check if your password is correctly stored by echoing it in the command prompt. (special characters may some times cause problems)

**Comment by khushhalgarg112** [comment]
You have to 

> After Turn on less secure app.............but still now it's not working.........what i do??

You have to required an app password 

https://khushhalgarg.hashnode.dev/send-emails-using-python-smtplib-module

follow this blog it explain how to get the password
